### DISCUSSION TITLE
Master list of Service Provider Types

### DISCUSSION THREAD

**Discussion by dsab123** [Azure/PSRule.Rules.Azure#1464] [Q&A] [answered]
Hello!

I am not using PSRule, but am interested in cobbling together a single "master" list of Azure `resources`, their `resourceTypes` and `apiVersions`.

The `types.json` files are exactly what I need to do this, but before I write a JSON parser I am wondering if there's already such a list in existence.

Any pointers or direction would be appreciated!

Thanks,
Daniel

**Comment by BernieWhite** [comment, marked as answer]
@dsab123 The providers and API versions come from the swagger / Open API specs for the Azure REST APIs. You can find that here: https://github.com/Azure/azure-rest-api-specs

Our current method, exports the details from the Azure REST APIs directly which may be incomplete based on the resource providers for a subscription that are registered but is suitable for our purposes and includes additional information that we need such as locations and availability zones.

You may prefer to go back to the specs for a full list. Also be aware that the list changes on a regular basis.

**Comment by dsab123** [comment]
Ah, that is very helpful @BernieWhite. Thank you!
